lib/utils/docker.go

```diff
@@ -1,38 +1,111 @@
 package utils
 
 import (
+	"bytes"
+	"context"
+	"encoding/json"
 	"log"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/client"
+	"github.com/sdslabs/katana/configs"
 )
 
 func BuildDockerImage(_ChallengeName string, _DockerfilePath string) {
-	log.Println("Building docker image, Please wait...")
-	cmd := "docker build -t " + "harbor.katana.local/katana/" + _ChallengeName + " " + _DockerfilePath
-	if err := RunCommand(cmd); err != nil {
-		log.Printf("Error: %s\n", err)
+	buf := new(bytes.Buffer)
+	if err := Tar(_DockerfilePath, buf); err != nil {
+		log.Fatal(err, ": error tarring directory")
+		return
+	}
+	ctx := context.Background()
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+	log.Println("Building Docker image, Please wait......")
```

Comment:
Give a line break before and after

Reply:
while printing this statement or in the code only ?

Reply:
Here, as well as other places you seem fit, it becomes more readable that way.